test.cmake

```diff
@@ -98,16 +98,23 @@ MACRO(ADD_PYTHON_UNIT_TEST NAME SOURCE)
   ADD_TEST(NAME ${NAME} COMMAND ${PYTHON_EXECUTABLE} "${PROJECT_SOURCE_DIR}/${SOURCE}")
   SET(PYTHONPATH)
 
-  SET(MODULE_PATH "${ARGN}")  # ARGN is not a variable
-  IF(MODULE_PATH)
+  SET(MODULES "${ARGN}")  # ARGN is not a variable
+  FOREACH(MODULE_PATH IN LISTS MODULES)
     LIST(APPEND PYTHONPATH "${CMAKE_BINARY_DIR}/${MODULE_PATH}")
-  ENDIF(MODULE_PATH)
-  
+  ENDFOREACH(MODULE_PATH IN LISTS MODULES)
+
   IF(DEFINED ENV{PYTHONPATH})
-    SET(PYTHONPATH "${PYTHONPATH}:$ENV{PYTHONPATH}")
+    LIST(APPEND PYTHONPATH "$ENV{PYTHONPATH}")
   ENDIF(DEFINED ENV{PYTHONPATH})
 
-  SET(ENV_VARIABLES "PYTHONPATH=${PYTHONPATH}")
+  # get path separator to join those paths
+  EXECUTE_PROCESS(COMMAND
```

Comment:
Is this line has to rely on Python?

Reply:
This is is the more portable way to do that, and Python is needed anyway at this point: `PYTHON_EXECUTABLE` is used in the first line of this macro.

We could hardcode the results, but I don't see the benefits.